src/backend/marsha/core/views.py

```diff
@@ -47,71 +48,98 @@ def serializer_class(self):
         """Return the serializer used by the view."""
 
     def get_context_data(self):
-        """Build a context with data retrieved from the LTI launch request.
+        """Build context for template rendering of configuration data for the frontend.
 
         Returns
         -------
         dictionary
-            context for template rendering:
+            context with configuration data for the frontend
+
+        """
+        try:
+            app_data = self._get_app_data()
+        except (LTIException, PortabilityError) as error:
+            logger.warning(str(error))
+            app_data = {
+                "state": "error",
+                "modelName": self.model.RESOURCE_NAME,
+                "resource": None,
+            }
+
+        context = {"app_data": json.dumps(app_data)}
+
+        if getattr(settings, "STATICFILES_AWS_ENABLED", False):
+            context["cloudfront_domain"] = settings.CLOUDFRONT_DOMAIN
+
+        return context
+
+    def _get_app_data(self):
+        """Build app data for the frontend with information retrieved from the LTI launch request.
+
+        Returns
+        -------
+        dictionary
+            Configuration data to bootstrap the frontend:
 
             For all roles
             +++++++++++++
 
-            - state: state of the LTI launch request. Can be one of `student`, `instructor` or
-                `error`.
-            - video: representation of the video including urls for the video file in all
-                resolutions, with thumbnails and timed text tracks.
+            - state: state of the LTI launch request. Can be one of `success` or `error`.
+            - modelName: the type of resource (video, document,...)
+            - resource: representation of the targetted resource including urls for the resource
+                file (e.g. for a video: all resolutions, thumbnails and timed text tracks).
 
             For instructors only
             ++++++++++++++++++++
 
-            - jwt_token: a short-lived JWT token linked to the video ID that will be
-                used as authentication.
+            - jwt_token: a short-lived JWT token linked to the resource ID that will be
+                used for authentication and authorization on the API.
 
         """
         lti = LTI(self.request, self.kwargs["uuid"])
-        try:
+        lti.verify()
+
+        app_data = None
+        if lti.is_student:
+            cache_key = (
+                f"app_data|{self.model.__name__:s}|{lti.get_consumer_site().domain:s}|"
```

Comment:
Just a comment, in all the project we use the `string.format` syntax nd not this one, we should maybe continue to use `string.format` ?

Reply:
I read that f strings are faster and I don't mind mixin. I agree that this is neglible and can revert to using format.

Reply:
I would prefer using f-strings everywhere!

Reply:
open an issue